packages/api/src/web/RouteRegistry.ts

```diff
@@ -44,18 +53,39 @@ export class RouteRegistry {
     requestPath: string,
     method: HttpMethod
   ): {
+    /**
+     * An awilix resolver for the actual route handler. If lookup fails,
+     * undefined.
+     */
     resolver?: Resolver<any> | ((...args: any[]) => any);
+    /**
+     * A map of the extracted path variables for the request (name => value).
+     * Undefined if lookup fails.
+     */
     pathVariables?: { [key: string]: string };
+    /**
+     * Information indicating why a lookup failed. Undefined if lookup
+     * succeeded.
+     */
     error?: "METHOD_NOT_SUPPORTED" | "ROUTE_NOT_SUPPORTED";
+    /**
+     * A list of allowed HTTP methods for the matched route. Undefined if the
+     * route is unsupported.
+     */
     allow?: HttpMethod[];
```

Comment:
If we just threw a UnknownRoute / UnknownMethod error and added a `allow` member, that might be cooler.